fury/ui/elements.py

```diff
@@ -3052,3 +3054,298 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+
+        Create a Canvas(Panel2D).
+        """
+        self.canvas = Panel2D(size=self.panel_size)
+        self.canvas.background.on_left_mouse_button_pressed = self.left_button_pressed
+        self.canvas.background.on_left_mouse_button_dragged = self.left_button_dragged
+
+        mode_data = {
+            "selection": ["selection.png", "selection-pressed.png"],
+            "line": ["line.png", "line-pressed.png"],
+            "quad": ["stop2.png", "quad-pressed.png"],
+            "circle": ["circle.png", "circle-pressed.png"],
+            "delete": ["cross.png", "cross-pressed.png"]
+        }
+
+        padding = 5
+        mode_panel_size = (len(mode_data) * 35 + 2 * padding, 40)
+        self.mode_panel = Panel2D(size=mode_panel_size, color=(0.5, 0.5, 0.5))
+        btn_pos = np.array([0, 0])
+
+        for mode, fname in mode_data.items():
+            icon_files = []
+            icon_files.append((mode, read_viz_icons(fname=fname[0])))
+            icon_files.append((mode+"-pressed", read_viz_icons(fname=fname[1])))
+            btn = Button2D(icon_fnames=icon_files)
+
+            def mode_selector(i_ren, _obj, btn):
+                self.update_button_icons(self.current_mode, btn.icon_names[0])
+                self.current_mode = btn.icon_names[0]
+                i_ren.force_render()
+
+            btn.on_left_mouse_button_pressed = mode_selector
+
+            self.mode_panel.add_element(btn, btn_pos+padding)
+            btn_pos[0] += btn.size[0]+padding
+
+        self.canvas.add_element(self.mode_panel, (0, 0))
+
+        self.mode_text = TextBlock2D(text="Select appropriate drawing mode using below icon")
+        self.canvas.add_element(self.mode_text, (0.0, 0.95))
+
+    def _get_actors(self):
+        """Get the actors composing this UI component."""
+        return self.canvas.actors
+
+    def _add_to_scene(self, scene):
+        """Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        scene : scene
+
+        """
+        self.current_scene = scene
+        self.iren = scene.GetRenderWindow().GetInteractor().GetInteractorStyle()
+        self.canvas.add_to_scene(scene)
+
+    def _get_size(self):
+        return self.canvas.size
+
+    def _set_position(self, coords):
+        """Set the lower-left corner position of this UI component.
+
+        Parameters
+        ----------
+        coords: (float, float)
+            Absolute pixel coordinates (x, y).
+        """
+        self.canvas.position = coords
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        pass
+
+    @property
+    def current_mode(self):
+        return self._current_mode
+
+    @current_mode.setter
+    def current_mode(self, mode):
+        self._current_mode = mode
+        if mode is not None:
+            self.mode_text.message = f"Mode: {mode}"
+        if self.iren is not None:
+            self.iren.force_render()
+
```

Comment:
Can you explain why you need to force render here ?

Reply:
Whenever the current mode was changed, the message in the`TextBlock2D` was also updated to show the current mode, at that time until the `force_render()` method wasn't called the new mode wasn't displayed (because internally `TextBlock2D` doesn't render whenever the message changes) So I had to do it manually.

But now as we update the button icons simultaneously, Now we don't need it.